xgcm/sgrid.py

```diff
@@ -0,0 +1,230 @@
+from collections import OrderedDict
+
+
+def assert_valid_sgrid(ds):
+    """Verify that the dataset meets SGRID conventions
+    We know this if the dataset has 'SGRID' listed in its conventions
+
+    Parameters
+    ----------
+    ds : xarray.dataset
+
+    Returns
+    ----------
+    True/False : boolean
+    """
+
+    if "SGRID" in ds.Conventions:
```

Comment:
This fails the docs build because you first need to check if `ds.Conventions` exists before you can safely check if "SGRID" is in there.

Reply:
Added a new commit that should deal with this.